src/test/java/com/pi4j/crowpi/components/SoundSensorComponentTest.java

```diff
@@ -0,0 +1,61 @@
+package com.pi4j.crowpi.components;
+
+import com.pi4j.crowpi.ComponentTest;
+import com.pi4j.io.gpio.digital.DigitalInput;
+import com.pi4j.io.gpio.digital.DigitalState;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class SoundSensorComponentTest extends ComponentTest {
+    protected SoundSensorComponent soundSensor;
+    protected DigitalInput din;
+
+    @BeforeEach
+    public void setUp() {
+        soundSensor = new SoundSensorComponent(pi4j);
+        din = soundSensor.getDigitalInput();
+    }
+
+    @Test
+    public void testGetSensorState() {
+        // given
+        SoundSensorComponent.SoundState resultState;
+
+        // when
+        resultState = soundSensor.getState();
+
+        // then
+        Assertions.assertEquals(SoundSensorComponent.SoundState.SILENT, resultState);
+    }
+
+    @Test
+    public void testIsNoise() {
+        // given
+        boolean result;
+
+        // when
+        result = soundSensor.isSilent();
+
+        // then
+        assertEquals(din.isHigh(), result);
+    }
```

Comment:
Once you are making use of `MockDigitalInput`, you could e.g. write two test methods `testNoise()` and `testSilence()`, the first one looking like this:

```java
    @Test
    public void testNoise() {
        // given
        din.mockState(DigitalState.HIGH);

        // when
        final boolean isSilent = soundSensor.isSilent();
        final boolean isNoisy = soundSensor.isNoisy();

        // then
        assertFalse(isSilent);
        assertTrue(isNoisy);
    }
```

Only by writing this example test locally I actually discovered that `isSilent()` and `isNoisy()` have inverted states for the digital input 😄 

Then we can actually test the full state mapping end-to-end and also ensure that the two states can not coexist. You can also check `getState()` in a similar way or introduce a `testUnknownState()` if you want to be able to fully drop the `testMapDigitalState` method.